### PR TITLE
Separate zkVote contract using agreed interface.

### PR DESCRIPTION
**Pull request by saiakilesh**
#24

### CODE REVIEW

File: scripts/committee.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ const app = command({
       type: string,
       description: "Nouns descriptor file location",
       long: 'nc_descriptor',
-      short: 'nc',
+      short: 'zkv',
```
[the remark this answers is not in the document]
Comment (dtebbs) [reply scripts/committee.ts:68]: (i.e. option name, filename, etc)

[the remark this answers is not in the document]
Comment (dtebbs) [reply scripts/committee.ts:68]: We may need to refactor the config files a bit.  Committee members will need to know the DKG and zkVote contracts (these could be in a single file).  Voters will need to know only the nouns contract address.

File: scripts/nouns/nouns.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import {
 } from "../crypto";
 import * as nouns_contract from "./nouns_contract";
 import * as dkg_contract from "./dkg_contract";
+import * as zkvote__contract from "./zkvote_contract";
```
Comment (dtebbs) [inline scripts/nouns/nouns.ts:7]: Typo:  `__` -> `_`.

File: scripts/nouns/nouns_contract.ts [hunk 3]
```diff
@@ -15,20 +14,11 @@ export type NounsContractDescriptor = {
 
 export async function deploy(
   deployer: Signer,
-  _dkg_address: string,
-  total_voting_power: bigint,
+  _zkVote: string,
```
Comment (dtebbs) [inline scripts/nouns/nouns_contract.ts:17]: Why not just `zkVote`?

File: scripts/nouns/voter.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ import {
 import { Nouns, NounsContractDescriptor } from "./nouns_contract";
 import * as nouns_contract from "./nouns_contract";
 import * as dkg_contract from "./dkg_contract";
+import * as zkvote__contract from "./zkvote_contract";
```
Comment (dtebbs) [inline scripts/nouns/voter.ts:8]: `zkvote_contract`

File: scripts/nouns/zkvote_contract.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import {
+    ZKVote, ZKVote__factory, NvoteVerifier__factory,
+    TallyVerifier__factory,
+  } from "../types";
+  import { Signer, Contract } from "ethers";
+  import { Provider } from "@ethersproject/providers";
+  export { ZKVote } from "../types";
+  
+  
+  export type ZKVoteContractDescriptor = {
```
Comment (dtebbs) [inline scripts/nouns/zkvote_contract.ts:10]: Doesn't have to be done now, but I think we can probably wind the dkg deployment into here, deploy both at the same time, and include the dkg contract address as another field in the descriptor.

Again, no need to do it now - we can just add it as an issue for later if you think it makes sense.

File: scripts/nouns/zkvote_contract.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import {
+    ZKVote, ZKVote__factory, NvoteVerifier__factory,
+    TallyVerifier__factory,
+  } from "../types";
+  import { Signer, Contract } from "ethers";
```
Comment (dtebbs) [inline scripts/nouns/zkvote_contract.ts:5]: Whitespace in this file seems off.  All statements from here are indented, trailing whitespace, etc.

**Review by dtebbs — COMMENTED** [review]
Good work.
Just a few points.  Plus I think the demo is also broken (possibly just contract deployment?).

File: scripts/nouns/nouns_contract.ts (hunk 3 above)
Comment (saiakilesh) [reply scripts/nouns/nouns_contract.ts]: Fixed.

File: contracts/interfaces/INounsPrivateVoting.sol [hunk 7]
```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MIT
+/// @title Interface for NounsPrivateVoting
+
+pragma solidity >=0.8.4;
+
+interface INounsPrivateVoting {
```
Comment (dtebbs) [inline contracts/interfaces/INounsPrivateVoting.sol:6]:
```suggestion
interface IZKVote {
```

File: contracts/interfaces/INounsPrivateVoting.sol [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+            uint256 proposalId, 
+            address voterAddress,
+            uint256 votingWeight,
+            uint[2][3] calldata voter_R_i, 
+            uint[2][3] calldata voter_M_i,
+            uint256[2] calldata proof_a,
+            uint256[2][2] calldata proof_b,
+            uint256[2] calldata proof_c
+    ) external;
+    function max_voting_power() external view returns (uint256);
```
Comment (dtebbs) [inline contracts/interfaces/INounsPrivateVoting.sol:21]:
```suggestion
    function maxTotalVotingWeight() external view returns (uint256);
```

File: contracts/interfaces/INounsDAOProxy.sol [hunk 9]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: MIT
+/// @title Interface for NounsDAOProxy
+
+pragma solidity >=0.8.4;
+
+interface INounsDAOProxy {
```
Comment (dtebbs) [inline contracts/interfaces/INounsDAOProxy.sol:6]:
```suggestion
interface IDAOProxy {
```

File: contracts/interfaces/INounsPrivateVoting.sol [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+pragma solidity >=0.8.4;
+
+interface INounsPrivateVoting {
+    function setupVote(
+            uint256 proposalId, 
+            uint256 endBlock
+    ) external;
+    function castPrivateVote(
+            uint256 proposalId, 
+            address voterAddress,
```
Comment (dtebbs) [inline contracts/interfaces/INounsPrivateVoting.sol:13]:
```suggestion
```

File: scripts/nouns/zkvote_contract.ts (hunk 5 above)
Comment (saiakilesh) [reply scripts/nouns/zkvote_contract.ts]: Will keep this for later, as suggested.

**Comment by saiakilesh** [comment]
@dtebbs Ready for re-review.

**Review by dtebbs — CHANGES_REQUESTED** [review]
Just a few small points.  Overall looks great.

**Pull request merged**